engine/src/main/java/io/camunda/zeebe/engine/Engine.java

```diff
@@ -130,9 +132,13 @@ public ProcessingResult process(
         return processingResultBuilder.build();
       }
 
-      final boolean isNotOnBlacklist =
-          !processingState.getBlackListState().isOnBlacklist(typedCommand);
-      if (isNotOnBlacklist) {
+      if (
```

Comment:
I would argue these particular conditions better fit into the engine as they are logic that decides "is a blacklist check needed at all, given the intent of the command". The existing condition inside the DBBlacklistState.isOnBlacklist however relates more to the following cast to extract the instanceid. I generally would argue for separating these to motivations from each other wdyt? happy to jump on a call for discussing this

Reply:
That way, we increase the coupling between `DBBlacklistState` and its caller classes. Every time you want to check if something is blacklisted, you need to implement your own logic to make sure the call will not fail. The ideal case would be to have the `canBeBlacklisted` static method inside the `DBBlacklistState` class and let clients decide if they want to prevent unnecessary DB calls. But if we will optimise this for all usages in the codebase, we can just make it a default behaviour.